src/main/java/frc/robot/subsystems/color_wheel/commands/FindColor.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems.color_wheel.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.color_wheel.ColorWheel;
+
+/**
+ * Spin the wheel until you reach the desired color.
+ */
+public class FindColor extends CommandBase {
+    private ColorWheel colorWheel;
+    private String color;
+    private double power;
+    private String tempColor;
+    private int targetColorDistance;//Amount of colors away from initial color to target color.
+    private int previousColor;
+    private boolean isNewColorSeen = false;
+    private int initColorIndex;
+    private int targetColorIndex;

```

Comment:
There is much easier way to create this function, you just need to update the sensor and move as long as we haven't seen the color.

Reply:
But then I don't get to check which path is faster and I don't get to control the speed as you get closer.

Reply:
You still can check the color at the beginning, and act accordingly. I'm sure that you can do it with less than 8 variables